### PR TITLE
feat(NotificationItem): Open internal notification links in same tab

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

NotificationItem: opens internal links in the same tab, external links in a new tab.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-894](https://codecademy.atlassian.net/browse/EGG-894)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/NotificationList/__tests__/NotificationItemNew-test.tsx [hunk 1]
```diff
@@ -54,6 +54,25 @@ describe('NotificationItem', () => {
...
+
+  it('opens an external link in a new tab', () => {
+    const { view } = renderView(getPropsByLink('https://help.codecademy.com'));
+    const href = view.getByRole('link').getAttribute('target');
+    expect(href).toStrictEqual('_blank');
+  });
+
   it('calls handleClick upon clicking a link', () => {
     const { view } = renderView(getPropsByLink(linkHref));
     view.getByRole('link').click();
```
Comment (katiezutter) [inline packages/gamut/src/NotificationList/__tests__/NotificationItemNew-test.tsx:78]: I'll change the name of this file to `NotificationItem-test.tsx` (and another file in the directory to `NotificationList-test.tsx` after review!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.6270e3.0<br/>@codecademy/gamut-labs@14.4.1-alpha.6270e3.0<br/>@codecademy/gamut@32.3.1-alpha.6270e3.0<br/>@codecademy/styleguide@36.4.1-alpha.6270e3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086d1d771bddb0edd378cc3--gamut-preview.netlify.app>https://6086d1d771bddb0edd378cc3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086d1d771bddb0edd378cc3>Deploy Logs</a></p>

File: packages/gamut/src/NotificationList/__tests__/NotificationItemNew-test.tsx (hunk 1 above)
Comment (saghdaey) [reply packages/gamut/src/NotificationList/__tests__/NotificationItemNew-test.tsx:78]: eep i must've missed that one when replacing the old components. 
Thanks!

**Review by saghdaey — APPROVED** [review]
✨

File: packages/gamut/src/NotificationList/NotificationItem.tsx [hunk 2]
```diff
@@ -54,6 +54,13 @@ const StyledImg = styled.img`
   width: 3rem;
 `;
 
+const isInternalLink = (href: string) => {
+  try {
+    return new URL(href).hostname === window.location.hostname;
```
Comment (sooinc) [inline packages/gamut/src/NotificationList/NotificationItem.tsx:59]: nice!

**Review by sooinc — APPROVED** [review]
LGTM! Thanks for adding in the tests!

File: packages/gamut/src/NotificationList/NotificationItem.tsx (hunk 2 above)
Comment (dreamwasp) [inline packages/gamut/src/NotificationList/NotificationItem.tsx:59]: could you link a monolith PR? this all looks good to me but wanna test in the monolith before ✅

Comment (katiezutter) [reply packages/gamut/src/NotificationList/NotificationItem.tsx:59]: Of course!! Here's the branch: https://github.com/codecademy-engineering/Codecademy/pull/24010, and the environment directly: https://pr-24010.dev-eks.codecademy.com

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by dreamwasp — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.52-alpha.77913d.0<br/>@codecademy/gamut-labs@14.4.4-alpha.77913d.0<br/>@codecademy/gamut@32.3.4-alpha.77913d.0<br/>@codecademy/styleguide@36.4.4-alpha.77913d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60883898b0b2f57d63c13d95--gamut-preview.netlify.app>https://60883898b0b2f57d63c13d95--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60883898b0b2f57d63c13d95>Deploy Logs</a></p>

**Pull request merged**